server/filestore.go

```diff
@@ -1556,6 +1573,9 @@ func (fs *fileStore) newMsgBlockForWrite() (*msgBlock, error) {
 	}
 
 	mb := &msgBlock{fs: fs, index: index, cexp: fs.fcfg.CacheExpire}
+	if len(fs.cfg.Subjects) == 1 {
```

Comment:
This is correct, however if we get an update we should re-evaluate this on the fs.lmb.